### PR TITLE
stats: reproduce redundant TYPE tags problem in Prometheus.

### PR DESCRIPTION
**Pull request by jmarantz**
Commit Message: Reproduces a scenario where it's difficult to use a streaming optimization for Prometheus stats based on scopes without introducing a bug. Context:

  * Issue that /stats/prometheus endpoint takes too much much memory: https://github.com/envoyproxy/envoy/issues/16139
  * Solution for non-Prometheus endpoints to use less memory for /stats and run faster: https://github.com/envoyproxy/envoy/pull/19693 which I believe is working well.
  * This solution mapped to Prometheus: https://github.com/envoyproxy/envoy/pull/24998 
  * A case where this solution has a duplicate`# TYPE` line due to two stats with the same tag-extracted-stat name from two different scopes: https://github.com/envoyproxy/envoy/issues/27173

Note that the existing unit tests did not exercise that scenario so this PR adds a testcase.

@rulex123 FYI
@zirain FYI

Additional Description:
Risk Level: low
Testing: just the new test
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

**Review by zirain — COMMENTED** [review]
LGTM

**Review by kyessenov — APPROVED** [review]

**Comment by jmarantz** [comment]
Thanks; this is just a new unit test so I don't think it needs senior maintainer approval.

**Pull request merged**
